pkg/operator/ceph/controller/spec.go

```diff
@@ -662,16 +681,36 @@ func PrivilegedContext(runAsRoot bool) *v1.SecurityContext {
 }
 
 // LogCollectorContainer runs a cron job to rotate logs
-func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Container {
-	return &v1.Container{
+func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) (*v1.Container, error) {
```

Comment:
Since we get yaml validation, no need to return an error here
```suggestion
func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) (*v1.Container) {
```